modules/createBrowserHistory.js

```diff
@@ -308,6 +308,19 @@ const createBrowserHistory = (props = {}) => {
     };
   };
 
+  const link = (path, state) => {
+    const currentLocation = {
```

Comment:
You're replication this code in each history implementation. Should it not be part of the `shouldReplace` method?

Reply:
Do you mean that `shouldReplace` should also call `push`/`replace`?

Reply:
No,I mean that you can move the following lines to the `shouldReplace` method:

```js
    const currentLocation = {
      path: history.location.pathname + history.location.search + history.location.hash,
      state: history.state || {}
    };
```

Reply:
I think not because depends on the history implementation you get this data from history or window.

Reply:
Then you could pass the history object to the method?

Reply:
Ey @guidobouman  take a look at refactors I did, thanks!